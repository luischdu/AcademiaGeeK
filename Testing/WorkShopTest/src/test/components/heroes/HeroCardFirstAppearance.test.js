import React from 'react'
import { shallow } from 'enzyme'
import { HeroCard } from '../../../components/heroes/HeroCard'
import '../../../setupTests'

describe('Pruebas con HeroCardFirstAppearance', () => {
    test('Validar primera apariciÃ³n de heroe', () => {
        const id = 'dc-batman'
        const superhero = 'Batman'
        const publisher = 'DC Comics'
        const alter_ego = 'Bruce Wayne'
        const first_appearance = 'Detective Comics #27'
        const characters = 'Bruce Wayne'

        const wrapper = shallow(
            <HeroCard
                id={id}
                superhero={superhero}
                publisher={publisher}
                alter_ego={alter_ego}
                first_appearance={first_appearance}
                characters={characters} />
        )

        const textParrafo = wrapper.find('small').text().trim()
        expect(wrapper).toMatchSnapshot()
        expect(textParrafo).toBe("Detective Comics #27")
        expect(wrapper.exists('.text-muted')).toBe(true)
    })

})