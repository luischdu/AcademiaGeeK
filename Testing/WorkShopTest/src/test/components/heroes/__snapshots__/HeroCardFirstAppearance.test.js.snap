// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con HeroCardFirstAppearance Validar primera aparición de heroe 1`] = `
<div
  className="card ms-3 animate__animated animate__fadeIn"
  style={
    Object {
      "maxWidth": 540,
    }
  }
>
  <div
    className="row no-gutters"
  >
    <div
      className="col-md-4"
    >
      <img
        alt="Batman"
        className="card-img"
        src="./assets/heroes/dc-batman.jpg"
      />
    </div>
    <div
      className="col-md-8"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
           
          Batman
           
        </h5>
        <p
          className="card-text"
        >
           
          Bruce Wayne
           
        </p>
        <p
          className="card-text"
        >
          <small
            className="text-muted"
          >
             
            Detective Comics #27
             
          </small>
        </p>
        <Link
          to="./hero/dc-batman"
        >
          Más...
        </Link>
      </div>
    </div>
  </div>
</div>
`;
